Cleanup some testing

SkafteNicki opened the pull request:
## What does this PR do?

`tests/functional` contained more or less code that had nothing to do with the functional back end.
PR moves code to more appropriate places.
Note: this also clean-up showed that `dice_score` metric is very poorly tested (now in its own file). Improved testing will follow in future PR.

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

Borda commented on tests/classification/test_precision_recall_curve.py:
```diff
@@ -119,3 +120,30 @@ def test_pr_curve(pred, target, expected_p, expected_r, expected_t):
+@pytest.mark.parametrize(
+    ["sample_weight", "pos_label", "exp_shape"],
+    [
+        pytest.param(1, 1.0, 42),
```
```suggestion
        (1, 1.0, 42),
```
If there is no pytest argument, let's use simple tuple

Borda commented:
Some are moved but for example classification is completely removed

SkafteNicki commented:
@Borda the `tests/functional/classification.py` file contained different unrelated parts:
1. some tests related to (classification) utilities: moved to `test_utilities.py`
2. some tests related to dice metric: moved to new file `test_dice.py` (in classification folder)
3. some tests related to function `_binary_clf_curve`: function is the main component of `PrecisionRecallCurve` metric and therefore move to the test file of that metric

tchaton reviewed (approved):
LGTM !
